File: v1/models/serviceAddedEvent.go
package models

import (
	"github.com/hypnospinner/plando/v1/models/abstract"
	"gorm.io/gorm"
)

// ServiceAddedEvent describes the service which client picked
// to be done in one's order
type ServiceAddedEvent struct {
	gorm.Model
	abstract.Event
	ID        uint
	ServiceID uint
	OrderID   uint
}

File: server/Plando/Models/Orders/ServiceAddedEvent.cs
using System;
using Plando.Common;
using Plando.Models.Services;

namespace Plando.Models.Orders
{
    public class ServiceAddedEvent : EventBase, IAggregator<Order>
    {
        public int ServiceId { get; set; }
        public int OrderId { get; set; }
        public OrderCreatedEvent OrderCreatedEvent { get; set; }
        public Order Push(Order aggregate)
        {
            if (aggregate is null)
                return null;

            aggregate.Id = OrderId;
            //aggregate.Services.Add()
            throw new NotImplementedException();
        }
    }
}